### PR TITLE
feat(select): enable separators in dropdown list items

### PR DESCRIPTION
**Pull request by Kiarokh**
fix https://github.com/Lundalogik/crm-feature/issues/4054

## Review:
- [ ] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

**Comment by Kiarokh** [comment]
it was a bit too complicated for me to solve

**Comment by Befkadu1** [comment]
I don't understand why the tests are failing? Any help?

**Comment by adrianschmidt** [comment]
> I don't understand why the tests are failing? Any help?

Loads of tests seem to fail because some value is `null`. Could it be that the test code can no longer find the element in the page or something like that?

**Comment by Befkadu1** [comment]
> Loads of tests seem to fail because some value is `null`. Could it be that the test code can no longer find the element in the page or something like that?

Yea that is what happened but I don't understand why the changes we made caused the issue

File: src/components/select/select.tsx
Comment (adrianschmidt) [inline src/components/select/select.tsx:1]: I'll have a look at the failing tests now 👍

Comment (adrianschmidt) [reply src/components/select/select.tsx:1]: Sorry, I got reminded by Pelle that there's a workflow failing, which I need to take a look at first. I'll get back to this as soon as I can though 👍

Comment (adrianschmidt) [reply src/components/select/select.tsx:1]: I'm back on this one now 👍

File: src/components/select/select.tsx [hunk 1]
```diff
@@ -94,7 +95,7 @@ export class Select {
      * List of options.
      */
     @Prop()
-    public options: Option[] = [];
+    public options: Array<Option | ListSeparator>;
```
Comment (adrianschmidt) [inline src/components/select/select.tsx:98]: You removed the default value here, and that's why the tests are failing.

```suggestion
    public options: Array<Option | ListSeparator> = [];
```

`fixup! feat(select): enable separators in dropdown list items`

Comment (Befkadu1) [reply src/components/select/select.tsx:98]: oh you are awesome! Have you done the fixup or should I do it?

Comment (adrianschmidt) [reply src/components/select/select.tsx:98]: I just added the suggestion. You can commit it straight here in the GitHub interface. I even wrote the commit message for you, so you can just copy and paste that into the dialog to make it a fixup 😄

Comment (Kiarokh) [reply src/components/select/select.tsx]: >  I never did a commit straight in GitHub.

It's the best feature of Gihub ;P you should learn it

File: src/components/select/examples/select-separators.tsx [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+@Component({
+    shadow: true,
+    tag: 'limel-example-select-with-separators',
+})
+export class SelectSeparatorsExample {
+    @State()
+    public value: Option;
+
+    private options: Array<Option | ListSeparator> = [
+        { separator: true, text: 'Alaskan / Hawaiian' },
```
[the remark this answers is not in the document]
Comment (Kiarokh) [reply src/components/select/examples/select-separators.tsx:22]: The intention was to showcase a "real world" example, where this feature can be useful and applicable. Please feel free suggest another one which is shorter :)

**Review by hannahu — APPROVED** [review]

**Review by jgroth — COMMENTED** [review]
API changes approved

**Review by jgroth — APPROVED** [review]

**Pull request merged**